### PR TITLE
[SPARK-24208][SQL][FOLLOWUP] Move test cases to proper locations

### PR DESCRIPTION
**Pull request by mgaido91**
## What changes were proposed in this pull request?

The PR is a followup to move the test cases introduced by the original PR in their proper location.

## How was this patch tested?

moved UTs

### CODE REVIEW

**Comment by mgaido91** [comment]
cc @gatorsmile @HyukjinKwon

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisSuite.scala [hunk 1]
```diff
@@ -557,4 +558,21 @@ class AnalysisSuite extends AnalysisTest with Matchers {
       SubqueryAlias("tbl", testRelation)))
     assertAnalysisError(barrier, Seq("cannot resolve '`tbl.b`'"))
   }
+
+  test("SPARK-24208: analysis fails on self-join with FlatMapGroupsInPandas") {
```
Comment (gatorsmile) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisSuite.scala]: please also remove the corresponding unit test case

Comment (mgaido91) [reply sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisSuite.scala]: sorry, I forgot that, I am doing it, thanks.

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisSuite.scala [hunk 2]
```diff
@@ -557,4 +558,21 @@ class AnalysisSuite extends AnalysisTest with Matchers {
...
+    val pythonUdf = PythonUDF("pyUDF", null,
```
Comment (gatorsmile) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisSuite.scala]: Nit: 
```
"pyUDF", 
null,
```

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisSuite.scala [hunk 3]
```diff
@@ -557,4 +558,21 @@ class AnalysisSuite extends AnalysisTest with Matchers {
...
+      StructType(Seq(StructField("a", LongType))),
+      Seq.empty,
+      PythonEvalType.SQL_GROUPED_MAP_PANDAS_UDF,
+      true)
```
Comment (gatorsmile) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisSuite.scala]: Nit: `udfDeterministic = true`

**Comment by mgaido91** [comment]
sure, I'll keep them in mind. Sorry for the mistakes, I'll be more careful. Thanks @gatorsmile.

**Pull request closed without merging**